pkg/kgo/sink.go

```diff
@@ -88,7 +88,6 @@ func (s *sink) createReq(id int64, epoch int16) (*produceRequest, *kmsg.AddParti
 		producerID:    id,
 		producerEpoch: epoch,
 
```

Comment:
Sorry even further - does `hasHookBatchWritten` even need to exist? It looks like it's only used for setting hasHook, which is now deleted